tests/unit/Store/CookieStoreTest.php

```diff
@@ -48,6 +49,23 @@ public function getMock(array $args = [])
         return $mockStore;
     }
 
+    /**
+     * Gain access to PHPUnit's mock invocation stack for analyzing calls.
+     * PHPUnit 8.4 removed the native getInvocations property, requiring this workaround.
+     *
+     * @param object $mock
+     *
+     * @return array
+     */
+    public function getMockInvocations(object $mock)
```

Comment:
Maybe too much work for this PR, but you could also configure the mocks with `->expects($this->exactly($count)`.

That also means we are no longer relying on an internal API, and removes a 'dirty' hack

Reply:
Agreed! That would be a much cleaner way of handling this test unit. Not a fan of the current hack around either. I'll make a note to rework that test in a separate PR. Thanks for the suggestion